edb/server/compiler_pool/pool.py

```diff
@@ -618,6 +619,16 @@ def get_debug_info(self):
     def get_size_hint(self) -> int:
         raise NotImplementedError
 
+    def _maybe_update_last_active_time(self):
+        if sys.exc_info()[0] is None:
+            self._last_active_time = time.monotonic()
+
+    async def health_check(self) -> bool:
+        if time.monotonic() - self._last_active_time > 10:
+            await self.make_compilation_config_serializer()
```

Comment:
possibly in a followup PR, doesn't need to be this one:

can we enforce a timeout on this (configurable through an env var would be nice, but hardcoded to 10 seconds or something would be fine too) so that if we time out this connection, we return a healthcheck failure message instead of letting upstream (Consul or whatever else is doing the healthchecks) hit its own request timeout